File: src/tests/fidl/source_compatibility/union-strict-flexible/llcpp/step_00_before.cc
// Copyright 2021 The Fuchsia Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include <fidl/fidl.test.unionstrictflexible/cpp/wire.h>  // nogncheck
namespace fidl_test = fidl_test_unionstrictflexible;

// [START contents]
void use_union(fidl_test::wire::JsonValue* value) {
  switch (value->Which()) {
    case fidl_test::wire::JsonValue::Tag::kIntValue:
      printf("int value: %d\n", value->int_value());
      break;
    case fidl_test::wire::JsonValue::Tag::kStringValue:
      printf("string value: %s\n", value->string_value().data());
      break;
  }
}
// [END contents]

int main(int argc, const char** argv) { return 0; }

File: src/tests/fidl/source_compatibility/enum-member-remove/go/step_00_before/step_00_before.go
// Copyright 2021 The Fuchsia Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

// +build !build_with_native_toolchain

package main

import (
	lib "fidl/fidl/test/enummemberremove"
)

// [START contents]
func writer(s string) lib.Color {
	switch s {
	case "blue":
		return lib.ColorBlue
	case "red":
		return lib.ColorRed
	case "yellow":
		return lib.ColorYellow
	default:
		return lib.Color_Unknown
	}
}

func reader(color lib.Color) string {
	switch color {
	case lib.ColorBlue:
		return "blue"
	case lib.ColorRed:
		return "red"
	case lib.ColorYellow:
		return "yellow"
	default:
		return "<unknown>"
	}
}

// [END contents]

func main() {}
